optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -0,0 +1,342 @@
+                        "Please ensure the best_configure is in model state dict!"
+                    )
+
+                try:
+                    inc_config = INCConfig.from_pretrained(model_id)
+                    if not is_torch_version("==", inc_config.torch_version):
+                        msg = f"Quantized model was obtained with torch version {inc_config.torch_version} but {_torch_version} was found."
+                        logger.warning(f"{msg}")
+                except Exception:
+                    logger.info("Couldn't verify torch version.")
+
+            model.eval()
+
+        return cls(
+            model,
+            config=config,
+            model_save_dir=model_save_dir,
+            use_cache=use_cache,
+            **kwargs,
+        )
+
+    @classmethod
+    def _from_transformers(
+        cls,
+        model_id: str,
+        config: PretrainedConfig,
+        use_auth_token: Optional[Union[bool, str]] = None,
+        revision: Optional[str] = None,
+        force_download: bool = False,
+        cache_dir: Optional[str] = None,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        use_cache: bool = True,
+        torch_dtype: Optional[Union[str, "torch.dtype"]] = None,
+        **kwargs,
+    ):
+        """
+        Export a vanilla Transformers model into a TorchScript model using `torch.jit.trace`.
+
+        Arguments:
+            model_id (`str` or `Path`):
+                The directory from which to load the model.
+                Can be either:
+                    - The model id of a pretrained model hosted inside a model repo on huggingface.co.
+                    - The path to a directory containing the model weights.            save_dir (`str` or `Path`):
+                The directory where the exported ONNX model should be saved, default to
+                `transformers.file_utils.default_cache_path`, which is the cache directory for transformers.
+            config (`PretrainedConfig`) :
+                an object of PretrainedConfig.
+            use_auth_token (`str` or `bool`):
+                Is needed to load models from a private repository
+            revision (`str`):
+                Revision is the specific model version to use. It can be a branch name, a tag name, or a commit id
+            kwargs (`Dict`, *optional*):
+                kwargs will be passed to the model during initialization
+        """
+        if is_torch_version("<", "2.0.0"):
+            raise ImportError("`torch>=2.0.0` is needed to trace your model")
+
+        task = cls.export_feature
+        model_kwargs = {
+            "revision": revision,
+            "use_auth_token": use_auth_token,
+            "cache_dir": cache_dir,
+            "subfolder": subfolder,
+            "local_files_only": local_files_only,
+            "force_download": force_download,
+            "torch_dtype": torch_dtype,
+        }
+
```

Comment:
Same it could make sense to call  `INCModel.from_pretrained() instead

Reply:
done